### DISCUSSION TITLE
Implementing actuator dynamics

### DISCUSSION THREAD

**Discussion by Dobid** [JSBSim-Team/jsbsim#998] [Q&A] [answered]
Hello again, in the goal of having a more realistic simulation of my UAV, I would like to add the actuator dynamics (elevon and throttle) described by those 2 transfer functions : 
![image](https://github.com/JSBSim-Team/jsbsim/assets/36502022/7eb386e0-90fc-4812-90b2-2abc56aa9406)
![image](https://github.com/JSBSim-Team/jsbsim/assets/36502022/59807c23-8461-43ba-b38b-fe0cda0886b3)
I went through the FGFCS documentation and I couldn't find a way to implement it. Is there a way to implement such transfer functions in the FDM file of my aircraft ?
Thanks in advance !

**Comment by jonsberndt** [comment, marked as answer]
This might help:

https://jsbsim.sourceforge.net/AutomaticFlightInJSBSim.pdf

**Reply by Dobid** [reply]
Awesome, thanks a lot !

**Comment by seanmcleod** [comment]
Yep, I was going to point you to the [JSBSim Reference Manual](https://jsbsim.sourceforge.net/JSBSimReferenceManual.pdf) which on page 46 starts describing the filter components. Looks like it's pretty much the same text included in Jon's Automatic Flight in JSBSim link above.

In addition to the filters also take a look at the actuator component.

https://github.com/JSBSim-Team/jsbsim/blob/056bb71fc432681cb72b84fd2a68d79360bb1064/src/models/flight_control/FGActuator.h#L54-L118

So in combination with the filter component and the actuator component you should be able to match your example above.
